kratos/python_scripts/apply_periodic_boundary_condition_process.py

```diff
@@ -0,0 +1,120 @@
+import KratosMultiphysics as KratosMultiphysics
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ApplyPeriodicBoundaryConditionProcess(Model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class ApplyPeriodicBoundaryConditionProcess(KratosMultiphysics.Process):
+    """This class is used in order to impose periodic boundary condition between two given modelparts which are related
+        via a rotation operation or a translation operation.
+
+    This class constructs the model parts containing the constraints that enforce the rigid body movement
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                        : "This process uses LinearMasterSlaveConstraint in order to impose a periodic boundary condition on the given submodelparts. The process takes the first provided submodelpart as master and the second as slave.",
+            "computing_model_part_name"   : "",
+            "model_part_name"             : "please_specify_model_part_name",
+            "first_model_part_name"       : "please_specify_model_part_name",
+            "second_model_part_name"      : "please_specify_model_part_name",
+            "interval"                    : [0.0, 1e30],
+            "transformation_settings"     : {
```

Comment:
These settings are not checked until they reach ApplyPeriodicConditionProcess. Therefore, these lines are code duplication and create a maintenance burden.
```suggestion
            "transformation_settings"     : {}
```